plugins/rollbar-backend/src/service/router.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { errorHandler } from '@backstage/backend-common';
+import { Logger } from 'winston';
+import Router from 'express-promise-router';
+import express from 'express';
+import { RollbarApi } from '../api';
+
+export interface RouterOptions {
+  rollbarApi?: RollbarApi;
+  logger: Logger;
+}
+
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const router = Router();
+  const logger = options.logger.child({ plugin: 'rollbar' });
+  const rollbarApi = options.rollbarApi;
+
+  if (rollbarApi) {
+    router.use(express.json());
+
+    const runAsync = createRunAsyncWrapper(logger);
+
+    router.get(
+      '/projects',
+      runAsync(async (_req, res) => {
+        const projects = await rollbarApi.getAllProjects();
+        res.status(200).header('').send(projects);
+      }),
+    );
+
+    router.get(
+      '/projects/:id',
+      runAsync(async (req, res) => {
+        const { id } = req.params;
+        const projects = await rollbarApi.getProject(id);
+        res.status(200).send(projects);
+      }),
+    );
+
+    router.get(
+      '/projects/:id/items',
+      runAsync(async (req, res) => {
+        const { id } = req.params;
+        const projects = await rollbarApi.getProjectItems(id);
+        res.status(200).send(projects);
+      }),
+    );
+
+    router.get(
+      '/projects/:id/top_active_items',
+      runAsync(async (req, res) => {
+        const { id } = req.params;
+        const query = req.query;
+        const items = await rollbarApi.getTopActiveItems(id, query as any);
+        res.status(200).send(items);
+      }),
+    );
+
+    router.get(
+      '/projects/:id/occurance_counts',
+      runAsync(async (req, res) => {
+        const { id } = req.params;
+        const query = req.query;
+        const items = await rollbarApi.getOccuranceCounts(id, query as any);
+        res.status(200).send(items);
+      }),
+    );
+
+    router.get(
+      '/projects/:id/activated_item_counts',
+      runAsync(async (req, res) => {
+        const { id } = req.params;
+        const query = req.query;
+        const items = await rollbarApi.getActivatedCounts(id, query as any);
+        res.status(200).send(items);
+      }),
+    );
+  }
+
+  router.use(errorHandler());
+
+  return router;
+}
+
+function createRunAsyncWrapper(logger: Logger) {
```

Comment:
Added to ensure that errors in async/await are logged properly. I have a feeling other plugins might suffer from exceptions not be propagated if they are using async/await with express. If that's the case, this could be promoted to a shared util function in core.

Reply:
This should be handled by the `express-promise-router`, are you sure it's needed?

Reply:
Tested again, and it doesn't seem to be propagating exceptions. I tested, by forcing the Rollbar api calls in `RollbarApi` to throw an exception by giving it an invalid token. Without this, I don't see any errors in the logger.

Looking at the [code for express-promise-router](https://github.com/express-promise-router/express-promise-router/blob/trunk/lib/express-promise-router.js) suggests it may not be. Probably something to be looked into. Going to keep it here for now, but I'll see if I can replicate this maybe in another plugin and open an issue if I can.

Reply:
Ok, let's keep it, but please create an issue!